index.js

```diff
@@ -1,121 +1,117 @@
-       */
-
-      rule.params = rule.params.replace(/\(\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*(<|>)(=?)\s*([a-z-]+)\s*(<|>)(=?)\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*\)/gi, function($0, $1, $2, $3, $4, $5, $6, $7) {
-
-        if (feature_name.indexOf($4) > -1) {
-          if ($2 === '<' && $5 === '<' || $2 === '>' && $5 === '>') {
-            var min = ($2 === '<') ? $1 : $7;
-            var max = ($2 === '<') ? $7 : $1;
-
-            // output differently depended on expression direction
-            // <mf-value> <|<= <mf-name> <|<= <mf-value>
-            // or
-            // <mf-value> >|>= <mf-name> >|>= <mf-value>
-            var equals_for_min = $3;
-            var equals_for_max = $6;
-
-            if ($2 === '>') {
-              equals_for_min = $6;
-              equals_for_max = $3;
-            }
-
-            return create_query($4, '>', equals_for_min, min) + ' and ' + create_query($4, '<', equals_for_max, max);
-          }
+  rule.params = rule.params.replace(/\(\s*([a-z-]+?)\s*([<>])(=?)\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*\)/gi, function($0, $1, $2, $3, $4) {
+    if (feature_name.indexOf($1) > -1) {
+      return create_query($1, $2, $3, $4);
+    }
+    //If it is not the specified attribute, don't replace
+    return $0;
+  })
+
+  /**
+   * 转换  <mf-value> <|<= <mf-name> <|<= <mf-value>
+   * 转换  <mf-value> >|>= <mf-name> >|>= <mf-value>
+   *   $1  $2$3 $4  $5$6  $7
+   * (500px <= width <= 1200px) => (min-width: 500px) and (max-width: 1200px)
+   * (500px < width <= 1200px) => (min-width: 501px) and (max-width: 1200px)
+   * (900px >= width >= 300px)  => (min-width: 300px) and (max-width: 900px)
+   */
+
+  rule.params = rule.params.replace(/\(\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*(<|>)(=?)\s*([a-z-]+)\s*(<|>)(=?)\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*\)/gi, function($0, $1, $2, $3, $4, $5, $6, $7) {
+
+    if (feature_name.indexOf($4) > -1) {
+      if ($2 === '<' && $5 === '<' || $2 === '>' && $5 === '>') {
+        const min = ($2 === '<') ? $1 : $7;
+        const max = ($2 === '<') ? $7 : $1;
+
+        // output differently depended on expression direction
+        // <mf-value> <|<= <mf-name> <|<= <mf-value>
+        // or
+        // <mf-value> >|>= <mf-name> >|>= <mf-value>
+        let equals_for_min = $3;
+        let equals_for_max = $6;
+
+        if ($2 === '>') {
+          equals_for_min = $6;
+          equals_for_max = $3;
         }
-        //如果不是指定的属性，不做替换
-        return $0;
-
-      });
 
-    });
+        return create_query($4, '>', equals_for_min, min) + ' and ' + create_query($4, '<', equals_for_max, max);
+      }
+    }
+    // If it is not the specified attribute, don't replace
+    return $0;
+  });
+}
+
+module.exports = () => ({
+  postcssPlugin: 'postcss-media-minmax',
+  AtRule(rule) {
+    if (rule.name !== "media" && rule.name !== "custom-media") {
+      return
+    }
 
-  }
+    transform(rule)
+  },
```

Comment:
[Migration guide](https://evilmartians.com/chronicles/postcss-8-plugin-migration#step-3-take-the-most-out-of-the-new-api) suggest subscribing to a specific names for better performance.

```js
  AtRule {
    media: (atRule) => {
      transform(atRule);
    },
    'custom-media': (atRule) => {
      transform(atRule);
    },
  },
```

Reply:
Updated

Reply:
@ai Can you help me see if there are other performance issues?